### PR TITLE
Align headers from v1.7.1 to v1.9.1

### PR DESCRIPTION
**Pull request by kcudnik**
Align enum values to match those from v1.9.1 so later on
branch v1.7.1 could be used to warm boot to v1.9.1

### CODE REVIEW

File: inc/sairouterinterface.h [hunk 1]
```diff
@@ -285,6 +285,15 @@ typedef enum _sai_router_interface_attr_t
...
     SAI_ROUTER_INTERFACE_ATTR_DISABLE_DECREMENT_TTL,
 
+    /**
+     * @brief Admin MPLS state
+     *
+     * @type bool
+     * @flags CREATE_AND_SET
+     * @default false
+     */
+    SAI_ROUTER_INTERFACE_ATTR_ADMIN_MPLS_STATE,
```
Comment (lguohan) [inline inc/sairouterinterface.h:295]: this is at the end, do we need them?

File: inc/saiqosmap.h [hunk 2]
```diff
@@ -77,6 +77,12 @@ typedef enum _sai_qos_map_type_t
     /** QOS Map to set traffic class and color to EXP */
     SAI_QOS_MAP_TYPE_TC_AND_COLOR_TO_MPLS_EXP = 0x0000000c,
 
+    /** QOS Map to set DSCP to Forwarding class */
+    SAI_QOS_MAP_TYPE_DSCP_TO_FORWARDING_CLASS = 0x0000000d,
+
+    /** QOS Map to set EXP to Forwarding class */
+    SAI_QOS_MAP_TYPE_MPLS_EXP_TO_FORWARDING_CLASS = 0x0000000e,
```
Comment (lguohan) [inline inc/saiqosmap.h:84]: this is the at the end, do we need to add them?

File: inc/saimpls.h [hunk 3]
```diff
@@ -199,6 +199,19 @@ typedef enum _sai_inseg_entry_attr_t
...
+     *
+     * When it is empty, then packet hits won't be counted
+     *
+     * @type sai_object_id_t
+     * @flags CREATE_AND_SET
+     * @objects SAI_OBJECT_TYPE_COUNTER
+     * @allownull true
+     * @default SAI_NULL_OBJECT_ID
+     */
+    SAI_INSEG_ENTRY_ATTR_COUNTER_ID,
```
Comment (lguohan) [inline inc/saimpls.h:213]: this is at the end, do we need them?

File: inc/saihostif.h [hunk 4]
```diff
@@ -427,6 +458,12 @@ typedef enum _sai_hostif_trap_type_t
      */
     SAI_HOSTIF_TRAP_TYPE_MPLS_ROUTER_ALERT_LABEL = 0x00008001,
 
+    /**
+     * @brief MPLS packets discarded due to label lookup miss
+     * (default packet action is drop)
+     */
+    SAI_HOSTIF_TRAP_TYPE_MPLS_LABEL_LOOKUP_MISS = 0x00008002,
```
Comment (lguohan) [inline inc/saihostif.h:465]: this is also at the end, do we need it?

File: inc/saihostif.h [hunk 5]
```diff
@@ -370,6 +382,25 @@ typedef enum _sai_hostif_trap_type_t
...
+     * router IP address (default packet action is drop)
+     */
+    SAI_HOSTIF_TRAP_TYPE_BFDV6_MICRO = 0x00004008,
+
+    /**
+     * @brief LDP traffic (TCP src port == 646 or TCP dst port == 646) to local
+     * router IP address or, (UDP dst port == 646) to the 'all routers on this
+     * subnet' group multicast address (224.0.0.2) (default packet action is drop)
+     */
+    SAI_HOSTIF_TRAP_TYPE_LDP = 0x00004009,
```
Comment (lguohan) [inline inc/saihostif.h:402]: this is also at the end, do we need it?

File: inc/saihostif.h [hunk 6]
```diff
@@ -322,6 +322,18 @@ typedef enum _sai_hostif_trap_type_t
...
+     * @brief Intermediate System-to-Intermediate System (IS-IS) protocol
+     *
+     * Traffic:
+     * L1 IS: 01:80:c2:00:00:14, All Level 1 Intermediate Systems Address
+     * L2 IS: 01:80:c2:00:00:15, All Level 2 Intermediate Systems Address
+     * All IS: 09:00:2b:00:00:05, All Intermediate System Network Entities address
+     *
+     * Default packet action is forward
+     */
+    SAI_HOSTIF_TRAP_TYPE_ISIS = 0x00002014,
```
Comment (lguohan) [inline inc/saihostif.h:335]: this is also at the end, do we need it?

File: inc/saidebugcounter.h [hunk 7]
```diff
@@ -308,6 +308,20 @@ typedef enum _sai_in_drop_reason_t
...
+    /**
+     * @brief SRV6 local SID drop
+     *
+     * Packet is dropped due to local SID configuration or incorrect value in SRV6 packet header
+     * e.g.: Local SID packet action is set to SAI_PACKET_ACTION_DROP
+     * Next Header is SRH and Segments Left value is 0 for End, End.X, End.T, End.B* endpoint types
+     * Next Header is not SRH while local SID is configured for packet DA
+     * Segments Left value is not 0 when received packet is destined to S and S is a local SID of type End.D*
+     */
+    SAI_IN_DROP_REASON_SRV6_LOCAL_SID_DROP,
```
Comment (lguohan) [inline inc/saidebugcounter.h:323]: these two attributes are at the end, do we need them?

File: inc/saihostif.h (hunk 6 above)
Comment (kcudnik) [inline inc/saihostif.h:335]: we also dont need those last at the sub ranges in 1.7

File: inc/saitunnel.h [hunk 8]
```diff
@@ -62,6 +62,18 @@ typedef enum _sai_tunnel_map_type_t
     /** TUNNEL Map Virtual Router ID to VNI */
     SAI_TUNNEL_MAP_TYPE_VIRTUAL_ROUTER_ID_TO_VNI = 0x00000007,
 
+    /** TUNNEL Map VSID to VLAN ID */
+    SAI_TUNNEL_MAP_TYPE_VSID_TO_VLAN_ID = 0x00000008,
```
Comment (lguohan) [inline inc/saitunnel.h:66]: this is also at the end, do we need it?

File: inc/saitunnel.h [hunk 9]
```diff
@@ -205,6 +217,24 @@ typedef enum _sai_tunnel_map_entry_attr_t
...
     SAI_TUNNEL_MAP_ENTRY_ATTR_VIRTUAL_ROUTER_ID_VALUE = 0x0000000d,
 
+    /**
+     * @brief VSID ID key
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @condition SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP_TYPE == SAI_TUNNEL_MAP_TYPE_VSID_TO_VLAN_ID or SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP_TYPE == SAI_TUNNEL_MAP_TYPE_VSID_TO_BRIDGE_IF
+     */
+    SAI_TUNNEL_MAP_ENTRY_ATTR_VSID_ID_KEY = 0x0000000e,
```
Comment (kcudnik) [inline inc/saitunnel.h:227]: we don't need them too probably since they are at the end

Comment (lguohan) [inline inc/saitunnel.h:227]: this is also at the end, do we need it?

File: inc/saidebugcounter.h (hunk 7 above)
Comment (kcudnik) [reply inc/saidebugcounter.h]: done

File: inc/saihostif.h (hunk 6 above)
Comment (kcudnik) [reply inc/saihostif.h]: done

File: inc/saihostif.h (hunk 5 above)
Comment (kcudnik) [reply inc/saihostif.h]: done

File: inc/saihostif.h (hunk 4 above)
Comment (kcudnik) [reply inc/saihostif.h]: done

File: inc/saimpls.h (hunk 3 above)
Comment (kcudnik) [reply inc/saimpls.h]: done

File: inc/sairouterinterface.h (hunk 1 above)
Comment (kcudnik) [reply inc/sairouterinterface.h]: done

File: inc/saiqosmap.h (hunk 2 above)
Comment (kcudnik) [reply inc/saiqosmap.h]: done

File: inc/saitunnel.h (hunk 9 above)
Comment (kcudnik) [reply inc/saitunnel.h]: done

File: inc/saitunnel.h (hunk 8 above)
Comment (kcudnik) [reply inc/saitunnel.h]: done

**Comment by rlhui** [comment]
@JaiOCP - would you please have someone in Broadcom to review this? Thanks.

File: inc/saihash.h [hunk 10]
```diff
@@ -54,120 +54,123 @@ typedef enum _sai_native_hash_field_t
...
+    SAI_NATIVE_HASH_FIELD_INNER_DST_IP = 0x00000003,
 
     /**
      * @brief Native hash field source IPv4.
      *
      * Also, refers to the outer source IPv4
      * in case for encapsulated packets
      */
-    SAI_NATIVE_HASH_FIELD_SRC_IPV4,
+    SAI_NATIVE_HASH_FIELD_SRC_IPV4 = 0x00000019,
```
Comment (tushar-ty) [inline inc/saihash.h:71]: How will this work for any upgrade from a previous version ?

Comment (kcudnik) [reply inc/saihash.h:71]: warm boot? will not be binary compatible with previous versions like v1.6

File: experimental/saiextensions.h [hunk 11]
```diff
@@ -37,7 +37,7 @@
 /**
  * @brief Extensions to SAI APIs
  *
- * @flags
+ * @flags free
```
Comment (rlhui) [inline experimental/saiextensions.h:40]: what does "free" mean here and why this is needed?

Comment (kcudnik) [reply experimental/saiextensions.h:40]: this is part of metadata, there are couple flags types, like strict, free, mixed etc, for more info analyze this PR https://github.com/opencomputeproject/SAI/pull/1314

**Comment by kcudnik** [comment]
closing this, as we already passed to v.113 version and we are backward compatible

**Pull request closed without merging**